metadata-service/factories/src/main/java/com/linkedin/gms/factory/entity/DatastaxAspectDaoFactory.java

```diff
@@ -0,0 +1,28 @@
+package com.linkedin.gms.factory.entity;
+
+import com.datastax.oss.driver.api.core.CqlSession;
+import com.linkedin.metadata.entity.datastax.DatastaxAspectDao;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.DependsOn;
+
+import javax.annotation.Nonnull;
+
+
+@Configuration
+public class DatastaxAspectDaoFactory {
+
+  @Autowired
+  ApplicationContext applicationContext;
```

Comment:
Was this causing issues? Directly using the ApplicationContext to retrieve beans is generally frowned upon. Really this shouldn't even need the @Autowired, you should be able to use method level injection i.e.:

protected DatastaxAspectDao createInstance(CqlSession cqlSession) {...}

Reply:
Good call. Done